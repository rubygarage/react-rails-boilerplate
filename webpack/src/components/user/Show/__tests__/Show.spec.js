import React from 'react';
import { shallow } from 'enzyme';
import User from 'components/user/Show';

describe('Show user', () => {
  const defaultProps = {
    user: {
      username: 'Boiler',
      email: 'example@boiler.com',
    },
    intl: {
      formatMessage: (...params) => (`intl.formatMessage(${JSON.stringify(params)})`),
    },
  };

  it('get info at the page', () => {
    const component = shallow(<User {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });

  it('empty info at the page', () => {
    const props = { ...defaultProps, user: {} };
    const component = shallow(<User {...props} />);
    expect(component).toMatchSnapshot();
  });
});
