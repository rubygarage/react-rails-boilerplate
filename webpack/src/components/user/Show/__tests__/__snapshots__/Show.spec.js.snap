// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show user empty info at the page 1`] = `
<main>
  <Connect(Header) />
  <div
    className="jumbotron"
  >
    <p
      className="lead"
    >
       This is User page 
    </p>
    <p>
      Username: 
    </p>
    <p>
      Email: 
    </p>
    <p>
      Avatar:
    </p>
    <p>
      none
    </p>
    <Link
      className="btn btn-primary"
      replace={false}
      to="/user/undefined/edit"
    >
      <FormattedMessage
        id="user.edit_user_data"
        values={Object {}}
      />
    </Link>
  </div>
</main>
`;

exports[`Show user get info at the page 1`] = `
<main>
  <Connect(Header) />
  <div
    className="jumbotron"
  >
    <p
      className="lead"
    >
       This is User page 
    </p>
    <p>
      Username: 
      Boiler
    </p>
    <p>
      Email: 
      example@boiler.com
    </p>
    <p>
      Avatar:
    </p>
    <p>
      none
    </p>
    <Link
      className="btn btn-primary"
      replace={false}
      to="/user/undefined/edit"
    >
      <FormattedMessage
        id="user.edit_user_data"
        values={Object {}}
      />
    </Link>
  </div>
</main>
`;
